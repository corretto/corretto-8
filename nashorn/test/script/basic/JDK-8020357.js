/*
 * Copyright (c) 2010, 2013, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

/**
 * JDK-8020357: Return range error for too big native array buffers
 *
 * @test
 * @run
 */

var UNSIGNED_INT_BITS = 31
var BYTES_PER_INT_32  =  4

var limit = Math.pow(2, UNSIGNED_INT_BITS)/BYTES_PER_INT_32

// A value over the limit should throw a RangeError.
try {
    new Int32Array(limit)
} catch(e) {
    print(e)
}
