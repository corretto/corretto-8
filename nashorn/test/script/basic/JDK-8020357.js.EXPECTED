RangeError: inappropriate array buffer length: 536870912
